### PR TITLE
Story 7

### PR DESCRIPTION
**Pull request by amhursh**
Related Issue: #21 

Summary: Reduce quantity of item in a cart functionality. Total price changes tested. Added price formatting.

Author(s): amhursh

### CODE REVIEW

File: spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb [hunk 1]
```diff
@@ -13,19 +13,75 @@
...
+
+    within(".item-#{item.id}") do
+      expect(page).to have_field("Quantity", with: "1")
+    end
+
+    within(".item-#{item2.id}") do
+      expect(page).to have_field("Quantity", with: "1")
+    end
+
+    expect(page).to have_content("Total: $#{(item2.formatted_price.to_f + item.formatted_price.to_f).to_s}")
```
Comment (randy-darsh) [inline spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb:56]: i like it.

File: app/models/cart.rb [hunk 2]
```diff
@@ -16,18 +16,22 @@ def update_item(id, quantity)
   def subtotal(item)
     id = item.id
     quantity = contents[id.to_s]
-    quantity * item.price
+    num = quantity * item.price
+    sprintf('%.2f', num)
```
Comment (limsammy) [inline app/models/cart.rb:20]: Absolutely love this method. Very powerful.

File: app/models/item.rb [hunk 3]
```diff
@@ -1,3 +1,7 @@
 class Item < ApplicationRecord
   belongs_to :category
+
+  def formatted_price
```
Comment (limsammy) [inline app/models/item.rb:4]: Great use of extracting logic and refactoring

File: spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb [hunk 4]
```diff
@@ -13,19 +13,75 @@
...
+
+# -As a visitor
+# -When I visit “/cart”
+# -Then I should see my item with a quantity of 1
+# -When I decrease the quantity
+# -Then my current page should be ‘/cart’
+# -And that item’s quantity should reflect the decrease
+# -And the subtotal for that item should decrease
+# -And the total for the cart should match that decrease
+
```
Comment (randy-darsh) [inline spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb:86]: probably want to erase this later

File: spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb [hunk 5]
```diff
@@ -13,19 +13,75 @@
...
-    expect(page).to have_content("Item Subtotal: $#{item.price}")
-    expect(page).to have_content("Total: #{item.price}")
+    expect(page).to have_content("Item Subtotal: $#{item.formatted_price}")
+    expect(page).to have_content("Total: $#{item.formatted_price}")
 
     fill_in "Quantity", with: "2"
     click_on "Update"
 
     expect(current_path).to eq(cart_path)
-    save_and_open_page
```
Comment (randy-darsh) [inline spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb:23]: remove the debugger

File: app/views/items/index.html.erb [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <div>
 	<% @items.each do |item| %>
-	<div class="item_card">
+	<div class="item_card index_item_<%= item.id %>">
```
Comment (limsammy) [inline app/views/items/index.html.erb:3]: I like the naming conventions for css classes so far.

File: spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb [hunk 7]
```diff
@@ -13,19 +13,75 @@
...
+
+    within(".item-#{item2.id}") do
+      expect(page).to have_field("Quantity", with: "1")
+      expect(page).to have_content("Item Subtotal: $#{item2.formatted_price}")
+    end
+
+    expect(page).to have_content("Total: $#{item2.formatted_price}")
   end
+
+# -As a visitor
```
Comment (limsammy) [inline spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb:78]: Refactor these comments out

File: app/views/items/index.html.erb (hunk 6 above)
Comment (JoelLindow) [reply app/views/items/index.html.erb:3]: This is pretty cool. Glad we figured this out.

**Comment by limsammy** [comment]
@cart, mind taking a look at this?

File: spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb (hunk 5 above)
Comment (JoelLindow) [reply spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb:23]: Yeah Randy. I think that's a great idea.

**Comment by JoelLindow** [comment]
This was a really good iteration. I think the added use of the Hyper terminal rocket allowed a much more pleasurable push to branch.

File: spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb (hunk 4 above)
Comment (JoelLindow) [reply spec/features/visitor/visitor_can_adjust_quantity_of_cart_items_spec.rb:86]: There's no probably about it.

**Comment by randy-darsh** [comment]
looks merging to dev branch

**Comment by cart** [comment]
@limsammy Haha apologies for missing this review.  I'll try to help out next time!

**Pull request merged**